src/core_algorithms.jl

```diff
@@ -90,30 +90,30 @@ end
 Performs a divide-and-conquer method similar to `dac_voronoi!` but has an additional site-elimination
 step which aims to reduce the work of subsequent steps.
 """
-function redac_voronoi!(grid::Matrix{T}, sites::Vector{T}; distance=euclidean, auxiliary=exact_aux) where {T<:SVector{2,Int}}
+function redac_voronoi!(grid::Matrix{T}, sites::Vector{T}; distance=euclidean, auxiliary=exact_aux) where {T<:Coord}
     TL = 1, 1
     BR = size(grid)
-    _redac_voronoi!(grid, TL, BR, EarlyStopper(sites); distance=distance, auxiliary=exact_aux)
+    _redac_voronoi!(grid, TL, BR, EarlyStopper(sites); distance=distance, auxiliary=auxiliary)
     return nothing
 end
 
 @inbounds function _redac_voronoi!(grid, TL, BR, sites::ES; distance, auxiliary) where {ES<:EarlyStopper}
     any(TL .> BR) && return nothing
     # Then, if the grid is a single cell then we are done
     if all(BR .== TL)
-        grid[TL...] = find_closest_site(TL, sites, distance=distance)
+        find_closest_site!(grid, TL, sites; distance=distance)
     elseif length(sites) == 1 # Same if there is a single site
         view(grid, TL[1]:BR[1], TL[2]:BR[2]) .= Ref(first(sites))
     else
         # Otherwise we check if all corners have the same closest site
         corners = get_corners(TL, BR)
-        closest_corners = (find_closest_site(corner, sites, distance=distance) for corner in corners)
+        closest_corners = (find_closest_site!(grid, corner, sites, distance=distance) for corner in corners)
         if allequal(closest_corners)
             view(grid, TL[1]:BR[1], TL[2]:BR[2]) .= Ref(first(closest_corners))
```

Comment:
I'm still not sure if this is enough for Chebyshev. Did you check with the randomized tests?

Reply:
No, but I would be surprised if it was not enough. I ran a bunch of randomised cases without integrating it into the test suite and they all passed. (Very scientific I know :'))